member/src/main/scala/com/improving/app/member/domain/Member.scala

```diff
@@ -56,220 +68,347 @@ object Member extends StrictLogging {
+        Left(StateError(useCommandValidationError(errors, s"${activateMemberCommand.productPrefix}")))
+    }
   }
-  def terminateMember(
-      memberId: MemberId,
-      terminatingMember: MemberId,
-      replyTo: ActorRef[StatusReply[MemberResponse]]
-  ): ReplyEffect[MemberEvent, MemberState] = {
-    val event = MemberTerminated(Some(memberId), Some(terminatingMember), Some(Timestamp(Instant.now(clock))))
-    Effect.persist(event).thenReply(replyTo) { _ => StatusReply.Success(MemberEventResponse(event)) }
 
+  private def suspendMember(
+    info: MemberInfo,
+    meta: MemberMetaInfo,
+    suspendMemberCommand: SuspendMember,
+  ): Either[StateError, MemberResponse] = {
+    MemberValidation.validateSuspendMemberCommand(suspendMemberCommand) match {
+      case Valid(_) =>
+        val newMeta = meta.copy(
+          lastModifiedBy = suspendMemberCommand.suspendingMember,
+          lastModifiedOn = Some(Timestamp(Instant.now(clock))),
+          currentState = StateEnum.MEMBER_STATUS_SUSPENDED
+        )
+        transitionMemberState(
+          info = info,
+          issuingCommand = suspendMemberCommand.productPrefix,
+          createEvent = () => {
+            MemberEventResponse(
+              MemberSuspended(
+                suspendMemberCommand.memberId,
+                Some(newMeta)
+              )
+            )
+          }
+        )
+      case Invalid(errors) =>
+        Left(StateError(useCommandValidationError(errors, s"${suspendMemberCommand.productPrefix}")))
+    }
   }
 
-  def updateMemberInfo(
-      memberId: MemberId,
-      memberInfo: MemberInfo,
-      actingMember: MemberId,
-      replyTo: ActorRef[StatusReply[MemberResponse]]
-  ): ReplyEffect[MemberEvent, MemberState] =
-    MemberValidation.validateMemberInfo(memberInfo) match {
-      case Valid(memberInfo) =>
-        val event =
-          MemberInfoUpdated(Some(memberId), Some(memberInfo), Some(actingMember), Some(Timestamp(Instant.now(clock))))
-        Effect.persist(event).thenReply(replyTo) { _ => StatusReply.Success(MemberEventResponse(event)) }
+  private def terminateMember(
+    meta: MemberMetaInfo,
+    terminateMemberCommand: TerminateMember
+  ): Either[StateError, MemberResponse] = {
+    MemberValidation.validateTerminateMemberCommand(terminateMemberCommand) match {
+      case Valid(_) =>
+        val newMeta = meta.copy(
+          lastModifiedBy = terminateMemberCommand.terminatingMember,
+          lastModifiedOn = Some(Timestamp(Instant.now(clock)))
+        )
+        val event = MemberTerminated(terminateMemberCommand.memberId, Some(newMeta))
+        Right(MemberEventResponse(event))
       case Invalid(errors) =>
-        Effect.reply(replyTo) {
-          StatusReply.Error(
-            s"Invalid Member Info: $memberInfo with errors: ${errors.map { _.errorMessage }.toList.mkString(",")}"
-          )
-        }
+        Left(StateError(useCommandValidationError(errors, s"${terminateMemberCommand.productPrefix}")))
```

Comment:
StateError was meant to express "This actor is in the wrong state to handle this message". There's ValidationError for "the message needs to be modified before it can be handled"